std/datetime/systime.d

```diff
@@ -8655,20 +8655,20 @@ public:
     static SysTime fromSimpleString(S)(in S simpleString, immutable TimeZone tz = null) @safe
         if (isSomeString!(S))
     {
-        import std.algorithm.searching : countUntil, find;
+        import std.algorithm.searching : find;
         import std.conv : to;
-        import std.string : strip;
+        import std.string : strip, indexOf;
 
-        auto dstr = to!dstring(strip(simpleString));
+        auto str = strip(simpleString);
 
-        auto spaceIndex = dstr.countUntil(' ');
+        auto spaceIndex = str.indexOf(' ');
         enforce(spaceIndex != -1, new DateTimeException(format("Invalid Simple String: %s", simpleString)));
 
-        auto found = dstr[spaceIndex + 1 .. $].find('.', 'Z', '+', '-');
-        auto dateTimeStr = dstr[0 .. $ - found[0].length];
+        auto found = str[spaceIndex + 1 .. $].find('.', 'Z', '+', '-');
```

Comment:
There's still auto-decoding happening here.

Reply:
The nightlies are currently down, but this still prints dchar for me:

```
> rdmddev --force --eval='pragma(msg, typeof(find("a", "b").front))'
dchar
```

Reply:
You'd probably have to do something like cast each needle to the character type of `str` to avoid it - either that or get `byCodeUnit` involved.

Reply:
@wilzbach @jmdavis Please see the opening post.

Reply:
An even better improvement at some point would be to do something like put any needles in static arrays with encode (if they're short enough to fit in the static arrays anyway) so that there's no need to do any decoding even when the needles don't have the same code unit type, but what you did does seem to fix this use case.